kentik_api_library/tests/integration/test_users.py

```diff
@@ -0,0 +1,57 @@
+import pytest
+
+from kentik_api import KentikAPI
+from kentik_api.public.errors import UnauthorizedError
+from kentik_api.public.user import User
+
+
+# TODO: table tests in pytest
+
+
+def test_create_user_fails_when_no_email_given() -> None:
+    client = KentikAPI("email@example.com", "api-test-token")
+    user = User(role="Member", email_service=True, email_product=True)
+
+    with pytest.raises(AssertionError, match="User email has to be provided"):
```

Comment:
I'm thinking of moving the requested object validation from users_api to users_payload (since the request payload objects know what fields are required and what are optional) and replacing assertions with more meaningful exception like RequestedObjectIncompleteError("User emial is required").
Same for the rest of *_api.
What you think?

Reply:
I agree. I was thinking of that and even added `ValidationError` for such purpose. We could do it in separate PR.